purchasing/opportunities/admin/views.py

```diff
@@ -199,6 +200,21 @@ def publish(opportunity_id):
         opportunity.is_public = True
         db.session.commit()
         flash('Opportunity successfully published!', 'alert-success')
+
+        opp_categories = [i.id for i in opportunity.categories]
+
+        vendors = Vendor.query.filter(
+            Vendor.categories.any(Category.id.in_(opp_categories))
+        ).all()
+
+        Notification(
```

Comment:
This will definitely need to be gated. We should only send these emails if `opportunity.is_advertised`. Otherwise, people will get emails as soon as a notification.